pkg/cli/error_formatter.go

```diff
@@ -0,0 +1,163 @@
+	case "yaml":
+		encoder := yaml.NewEncoder(os.Stderr)
+		encoder.SetIndent(2)
+		_ = encoder.Encode(errorOutput)
+		encoder.Close()
+	}
+}
+
+// formatTextError outputs errors in human-readable format
+func formatTextError(err error, verbose bool) {
+	if verbose {
+		fmt.Fprintf(os.Stderr, "Error: %+v\n", err)
+	} else {
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+	}
+}
+
+// mapErrorToStruct converts Go errors to structured Error format
+func mapErrorToStruct(err error, verbose bool) Error {
+	errorStruct := Error{
+		Message:  err.Error(),
+		ExitCode: bkErrors.GetExitCodeForError(err),
+	}
+
+	// Map specific error types to codes and hints
+	switch {
+	case isAuthError(err):
+		errorStruct.Code = "AUTH_REQUIRED"
+		errorStruct.Hint = "Run 'bk configure' to set up authentication"
+		errorStruct.Recoverable = true
+	case isNotFoundError(err):
+		errorStruct.Code = "RESOURCE_NOT_FOUND"
+		errorStruct.Hint = "Verify the resource exists and you have access to it"
+		errorStruct.Recoverable = true
+	case isNetworkError(err):
+		errorStruct.Code = "NETWORK_ERROR"
+		errorStruct.Hint = "Check your internet connection and try again"
+		errorStruct.Recoverable = true
+	case isConfigError(err):
+		errorStruct.Code = "CONFIG_ERROR"
+		errorStruct.Hint = "Check your configuration with 'bk configure'"
+		errorStruct.Recoverable = true
+	case isValidationError(err):
+		errorStruct.Code = "VALIDATION_ERROR"
+		errorStruct.Hint = "Check the command arguments and try again"
+		errorStruct.Recoverable = true
+	default:
+		errorStruct.Code = "GENERAL_ERROR"
+		errorStruct.Recoverable = false
+	}
+
+	return errorStruct
+}
+
+// Helper functions to identify error types
+func isAuthError(err error) bool {
+	// Check if error is related to authentication
+	msg := err.Error()
+	return contains(msg, "authentication") ||
+		contains(msg, "unauthorized") ||
+		contains(msg, "token") ||
+		contains(msg, "401")
+}
+
+func isNotFoundError(err error) bool {
+	// Check if error is related to resource not found
+	msg := err.Error()
+	return contains(msg, "not found") ||
+		contains(msg, "404") ||
+		contains(msg, "does not exist")
+}
+
+func isNetworkError(err error) bool {
+	// Check if error is related to network issues
+	msg := err.Error()
+	return contains(msg, "connection") ||
+		contains(msg, "network") ||
+		contains(msg, "timeout") ||
+		contains(msg, "dial")
+}
+
+func isConfigError(err error) bool {
+	// Check if error is related to configuration
+	msg := err.Error()
+	return contains(msg, "config") ||
+		contains(msg, "configuration") ||
+		contains(msg, "organization not set")
+}
+
+func isValidationError(err error) bool {
+	// Check if error is related to validation
+	msg := err.Error()
+	return contains(msg, "invalid") ||
+		contains(msg, "required") ||
+		contains(msg, "validation") ||
+		contains(msg, "format")
+}
+
+// contains checks if a string contains a substring (case-insensitive)
+func contains(s, substr string) bool {
```

Comment:
Wouldn't something like `strings.Contains(strings.ToLower(s), substr)` be enough?